src/index.js

```diff
@@ -0,0 +1,159 @@
+export default class toColor {
+  HUE_MAX = 360;
+
+  constructor(seed, options) {
+    this.options = options || {};
+    this.seed = typeof seed === 'string' ? this._stringToInteger(seed) : seed;
+    this.known = [];
+  }
+
+  getColor() {
+    const { distance, knownMax } = this.options;
+    const h = this._pickHue();
+    const s = this._pickSaturation(h);
+    const b = this._pickBrightness(h, s);
+    const hsl = this._HSVtoHSL(h, s, b);
+    const PASSABLE_DISTANCE = typeof distance !== 'undefined' ? distance : 37; 
+    const KNOWN_MAX = typeof knownMax !== 'undefined' ? knownMax : 20; 
+
+    // Detect color similarity. If values are too close to one another, call
+    // getColor until enough dissimilarity is acheived.
+    if (this.known.length &&
+      this.known.length <= KNOWN_MAX &&
+      this.known.some(v => this._euclideanDistance(v, hsl) < PASSABLE_DISTANCE)) {
+      return this.getColor();
```

Comment:
We should guard against stack overflows. One way to do that is to pass a counter to getColor, then: 
- If counter hits a limit, ignore passable distance
- Gradually decrement passable distance with each iteration

Another thing to consider is to gradually decrement passable distance as known size increases, or reset our known array to empty every time known length hits max. The 20 known max limits feels arbitrary, and making this a customizable param feels unecessary. We are handing the user a trap by telling them they _could_ use a max of 10,000 – it just won't work.

Reply:
Good points. I like the idea of decrementing passable distance as known size increases 👍